sda-auth/info.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/kataras/iris/v12"
+	"github.com/neicnordic/crypt4gh/keys"
+	log "github.com/sirupsen/logrus"
+)
+
+type Info struct {
+	ClientID  string `json:"client_id"`
+	OidcURI   string `json:"oidc_uri"`
+	PublicKey string `json:"public_key"`
+	InboxURI  string `json:"inbox_uri"`
+}
+
+func readPublicKeyFile(filename string) (key *[32]byte, err error) {
+	log.Info("Reading Public key file")
+	file, err := os.Open(filepath.Clean(filename))
+	if err != nil {
+		return nil, err
+	}
+
+	publicKey, err := readPublicKey(file)
+	if err != nil {
+		return nil, fmt.Errorf(err.Error()+": %s", filename)
+	}
+
+	return &publicKey, err
+}
+
+// Wrapper for the respective crypt4gh function. Since this function panics if the key is
+// malformed, so we handle that as well as errors.
+func readPublicKey(reader io.Reader) (key [32]byte, err error) {
+
+	defer func() {
+		if recover() != nil {
+			err = fmt.Errorf("malformed key file")
```

Comment:
Possibly nice to report any error message as well?

Reply:
This error would be returned, right? It also comes from "github.com/neicnordic/crypt4gh/blob/master/keys"

Reply:
Yes, I didn't think/look correctly (but crypt4gh throwing a panic really should be fixed, will try to look at that soonish).